source/endpoint-reference/endpoint-folders.rst

```diff
@@ -0,0 +1,191 @@
+Variable Folders
+----------------
+
+In order to organize number of variables in datasets, Crunch provides a
+hierarchical structure to group variables called "Folders".
+
+Much like the variable hierarchical order, folders allow to have nested groups
+containing more groups or dataset variables.
+
+Unlike the variables' order, a variable can only exist in one folder at the time.
+
+Root
+-----
+
+``/datasets/{id}/folders/``
+
+Returns a Shoji catalog containing the root folder of the hierarchy.
+
+When a dataset is created without any order specified, there will not be any
+folder associated to any of its variables so the variable folders' root will
+only contain a flat list of all the existing dataset variables.
+
+As variables get grouped into folders the folders root will continue showing
+the ungrouped variables and existing folders.
+
+
+Subfolder endpoint
+~~~~~~~~~~~~~~~~~~
+
+``/datasets/{id}/folders/{folder_id}/``
+
+All subfolders' URLs follow straight from the folders' root URL regardless
+of their nested location in the folders' tree.
+
+All subfolders' payload has the same shape as the root catalog.
+
+Folder payload
+--------------
+
+All folders, trash and root return a similar Shoji catalog with the same shape
+which includes the following members:
+
+======== ======= ===============================================================
+Member   Type    Description
+======== ======= ===============================================================
+size     int     Indicates the number of total variables stored under this
+                 folder and its subfolders.
+-------- ------- ---------------------------------------------------------------
+body     object  Contains a `name` attribute, which is the folder's name.
+                 The root folder has an empty string as name.
+-------- ------- ---------------------------------------------------------------
+index    object  Behaves like a standard Shoji catalog containing the children
+                 of the folder keyed by URL. This catalog will contain a mixed
+                 set of folder and variables URLs. Clients should differentiate
+                 each of the objects by looking at the tuple's `type` attribute.
+                 Folders will have type "folder" while variables will contain
+                 their variables type.
+-------- ------- ---------------------------------------------------------------
+graph    list    A flat list containing the list of URLs of variables and
+                 subfolders on the given folder.
+======== ======= ===============================================================
+
+JSON example:
+
+.. language_specific::
+   --JSON
+   .. code:: json
+
+      {
+          "element": "shoji:catalog",
+          "self": "https://app.crunch.io/api/datasets/5ee0a0/variables/",
```

Comment:
shouldn't this say folders/ instead of variables/? etc below

Reply:
True